frontend/src/hooks/useInput/useInput.ts

```diff
@@ -1,12 +1,14 @@
 import type { ChangeEvent } from 'react';
 import { useState } from 'react';
 
+import type { UseInputResult } from './useInput.type';
+
 interface ValidationRules {
   pattern?: RegExp;
   errorMessage?: string;
 }
 
-const useInput = (rules?: ValidationRules) => {
+const useInput = (rules?: ValidationRules): UseInputResult => {
```

Comment:
오, 유틸리티타입 적극적으로 활용해 보는 것 좋네요. 다만, 커스텀 훅을 정의할 때 해당 커스텀 훅의 반환값 타입을 명시하면 함수가 함수의 반환값 타입을 참조하는 자기 참조 문제가 있어, 커스텀 훅을 정의한 후 아래에서 ReturnType을 활용해 보도록 할게요~
```ts
export type UseInputResult = ReturnType<typeof useInput>;
```
아래는 참고하시면 좋을 것 같아, 에러 메시지를 함께 첨부합니다
```
'useInput' implicitly has type 'any' because it does not have a type annotation 
and is referenced directly or indirectly in its own initializer.
```

Reply:
그리고, ReturnType을 활용하는 만큼 타입 이름도 ~Return으로 수정해서 사용하는 유틸리티타입이랑 이름을 맞춰보도록 할게요~